prow/cmd/initupload/main.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"io/ioutil"
+	"path"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/option"
+
+	"k8s.io/test-infra/prow/pjutil"
+	"k8s.io/test-infra/prow/pod-utils/clone"
+	"k8s.io/test-infra/prow/pod-utils/gcs"
+)
+
+var (
+	cloneLog = flag.String("clone-log", "", "path to the output file for the cloning step")
```

Comment:
Personally I feel like we should avoid using globals and putting flags in globals as it encourages lazy code. I prefer something like a `func flagOptions() options {}` [pattern](https://github.com/kubernetes/test-infra/blob/master/robots/commenter/main.go#L57) instead.

Reply:
Happy to tackle this in a follow-up, pretty much every entrypoint in Prow uses the `var()` pattern right now.